### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by Dangelo-JAN**
# In this mileston i:

1. Used the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

2. In the BookContainer file I did:

- Define action types for adding and removing a book.
- Set the initial state to be an empty array of books.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- Define state changes for the actions that you created.
- In case of unknown action - return the current state.

3. In the Categories file I did:

- Define an action type for checking the status.
- Set the initial state to be an empty array of categories.
- Export Action Creators for your actions.
- Write your reducer and export it as default.
- For the check status action return a string "Under construction".
- In case of unknown action - return the current state.
- Configure the Redux Store (/src/redux/configureStore.js):

4. In the configureStore file I did:

- Importing the necessary methods from Redux.
- Combine both reducers

### CODE REVIEW

File: src/redux/BookList/BookList.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+// Reducer
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+    // do reducer stuff
+    case CREATE:
+      return { ...state };
+
+    case REMOVE:
+      return { ...state };
```
Comment (keroloussamy) [inline src/redux/BookList/BookList.js]: - [x] Kindly on the `CREATE` action you have to add to your list, something like ` return [...state, action.newBook ];` to add the new book to the existing ones.
- [x] Kindly on the `REMOVE` action you have to remove from your list, here you can use the `filter` method to remove the book from your list.

File: src/redux/Categories/Categories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const CHECK_STATUS = 'react-bookstore-app/categories/CHECK_STATUS';
+const initialState = {};
+
+// Reducer
+export default function reducer(state = initialState, action) {
+  switch (action.type) {
+    // do reducer stuff
+    case CHECK_STATUS:
+      return { ...state };
```
Comment (keroloussamy) [inline src/redux/Categories/Categories.js]: - [x] Kindly on the `CHECK_STATUS` action you have to return `Under construction` as the requirements say.

**Review by keroloussamy — CHANGES_REQUESTED** [review]
Hi @Dangelo-JAN, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️

## Required Changes ♻️
_Check the comments under the review._




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APPROVED :green_circle: :clap: 

Hello team,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

TO HIGHLIGHT:

- Your project is professional 
- Your repo is professional 
- Well structured files 


Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

well-1

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**